src/transformers/models/informer/modeling_informer.py

```diff
@@ -858,31 +876,21 @@ def forward(
         else:
             attn_weights_reshaped = None
 
-        # The authors didn't use attention dropout.
-        # Not removing this yet, waiting for Kashif approval
-        # attn_probs = nn.functional.dropout(attn_weights, p=self.dropout, training=self.training)
-        # attn_output = torch.bmm(attn_probs, value_states)
-        attn_output = torch.bmm(attn_weights, value_states)
-
-        # Build final output
-        # reimplemented from the original:
-        # https://github.com/zhouhaoyi/Informer2020/blob/ac59c7447135473fb2aafeafe94395f884d5c7a5/models/attn.py#L70
-        if not self.is_decoder:
-            v_aggregated = value_states.mean(dim=1)
-            v_aggregated = v_aggregated.unsqueeze(dim=1).expand(bsz * self.num_heads, L_Q, v_aggregated.size(-1))
-        else:
-            v_aggregated = value_states.cumsum(dim=1)
+        attn_probs = nn.functional.dropout(attn_weights, p=self.dropout, training=self.training)
 
-        # https://github.com/zhouhaoyi/Informer2020/blob/ac59c7447135473fb2aafeafe94395f884d5c7a5/models/attn.py#L90
-        dim_for_slice = torch.arange(v_aggregated.size(0)).unsqueeze(-1)
-        v_aggregated[dim_for_slice, M_top, :] = attn_output
+        # get initial context
```

Comment:
What about the case it's needed to calculate the mean?
```python
if not self.is_decoder:
            v_aggregated = value_states.mean(dim=1)
```

Reply:
fixed thanks!